torch/distributed/tensor/_api.py

```diff
@@ -270,23 +270,13 @@ def __new__(
         # new method instruct wrapper tensor from local_tensor and add
         # placement spec, it does not do actual distribution
         assert spec.tensor_meta is not None, "TensorMeta should not be None!"
-        # _dispatch_key_set_with_conjugate_and_negative_corresponding_to_tensor exists
-        # to support this case, because manipulating DispatchKeySet via pybind11 is
-        # disappointingly slow. Reduces DTensor overhead as observed in perf by a couple
-        # percent. (This still pybinds one DispatchKeySet; it would be even faster to
-        # fuse _make_wrapper_subclass with this operation, but that's more invasive.)
-        extra_dispatch_keys = torch._C._dispatch_key_set_with_conjugate_and_negative_corresponding_to_tensor(
-            local_tensor
-        )
-        r = torch.Tensor._make_wrapper_subclass(
+
+        r = torch.Tensor._make_dtensor(
```

Comment:
@bdhirsh does this change also allow us to remove the torch._disable_dynamo annotation on this function? I'm still pretty unfamiliar with dynamo

Reply:
although that reminds me - @anijain2305 I remember you telling me a long time ago that if we wanted a python function to be disabled with `recursive=True`, we have no way of doing that in dynamo and we're forced to do it via decorator on the python function itself.

We care a lot about DTensor overhead now, and this decorator is slow. If we're willing to hardcode some DTensor knowledge in dynamo, what do you think about effectively generated the `_disable_dynamo()` decorator around `DTensor.__new__` here in dynamo?